pkg/evetestkit/utils.go

```diff
@@ -0,0 +1,487 @@
+	}
+	appStatesSlice := make([]*eve.AppInstState, 0, len(state.Applications()))
+	appStatesSlice = append(appStatesSlice, state.Applications()...)
+	for _, app := range appStatesSlice {
+		if app.Name == appName {
+			return app.EVEState, nil
+		}
+	}
+
+	return "", fmt.Errorf("app %s not found", appName)
+}
+
+// SSHExec executes a command on the app VM running on the EVE node.
+func (node *EveNode) SSHExec(appName, command string) (string, error) {
+	appConfig := node.getAppConfig(appName)
+	if appConfig == nil {
+		return "", fmt.Errorf("app %s not found, make sure to deploy app/vm with WithSSH option", appName)
+	}
+
+	host := fmt.Sprintf("%s:%s", node.ip, appConfig.sshPort)
+
+	config := &ssh.ClientConfig{
+		User: appConfig.sshUser,
+		Auth: []ssh.AuthMethod{
+			ssh.Password(appConfig.sshPass),
+		},
+		HostKeyCallback: ssh.InsecureIgnoreHostKey(),
+		Timeout:         30 * time.Second,
+	}
+	client, err := ssh.Dial("tcp", host, config)
+	if err != nil {
+		return "", fmt.Errorf("failed to dial: %s", err)
+	}
+	defer client.Close()
+
+	session, err := client.NewSession()
+	if err != nil {
+		return "", fmt.Errorf("failed to create session: %s", err)
+	}
+	defer session.Close()
+
+	output, err := session.CombinedOutput(command)
+	if err != nil {
+		return "", fmt.Errorf("failed to run command: %s", err)
+	}
+
+	return string(output), nil
+}
+
+// SCPCopy copies a file from the local machine to the app VM running on the EVE node.
+func (node *EveNode) SCPCopy(appName, localFile, remoteFile string) error {
+	appConfig := node.getAppConfig(appName)
+	if appConfig == nil {
+		return fmt.Errorf("app %s not found, make sure to deploy app/vm with WithSSH option", appName)
+	}
+
+	host := fmt.Sprintf("%s:%s", node.ip, appConfig.sshPort)
+
+	config := &ssh.ClientConfig{
+		User: appConfig.sshUser,
+		Auth: []ssh.AuthMethod{
+			ssh.Password(appConfig.sshPass),
+		},
+		HostKeyCallback: ssh.InsecureIgnoreHostKey(),
+		Timeout:         30 * time.Second,
+	}
+	client, err := ssh.Dial("tcp", host, config)
+	if err != nil {
+		return fmt.Errorf("failed to dial: %s", err)
+	}
+	defer client.Close()
+
+	session, err := client.NewSession()
+	if err != nil {
+		return fmt.Errorf("failed to create session: %s", err)
+	}
+	defer session.Close()
+
+	err = scp.CopyPath(localFile, remoteFile, session)
+	if err != nil {
+		return nil
+	}
+
+	return nil
+}
+
+// WithSSH is an option that sets the SSH configuration for the app running on
+// the EVE node, this should be use with DeployVM function.
+func WithSSH(user, pass, port string) VMOption {
+	return func(n *EveNode) {
+		// slices are ordered, so the last element is the one
+		// we're going to set things for.
+		n.apps[len(n.apps)-1].sshUser = user
+		n.apps[len(n.apps)-1].sshPass = pass
+		n.apps[len(n.apps)-1].sshPort = port
+	}
+}
+
+// DeployVm deploys a VM on the EVE node
+func (node *EveNode) DeployVM(imageURL string, pc openevec.PodConfig, options ...VMOption) error {
```

Comment:
Btw. if this is not limited to VMs but can be used for containers, then maybe `DeployApp` is a better name

Reply:
I haven't tested containers, but I guess it should work with a container pod config. I rename it.